LLama.KernelMemory/LLamaSharpTextEmbeddingGenerator.cs

```diff
@@ -89,5 +92,15 @@ public Task<IList<ReadOnlyMemory<float>>> GenerateEmbeddingsAsync(IList<string>
 
             return Task.FromResult(results);
         }
+
+        /// <inheritdoc/>
+        public Task<Embedding> GenerateEmbeddingAsync(string text, CancellationToken cancellationToken = default)
+        {
+            var embeddings = _embedder.GetEmbeddings(text);
+            return Task.FromResult(new Embedding(embeddings));
+        }
+
+        /// <inheritdoc/>
+        public int CountTokens(string text) => _embedder.Tokenize(text).Length;
```

Comment:
```suggestion
        public int CountTokens(string text) => _weights.NativeHandle.Tokenize(text).Length;
```

Doing this means the `Tokenize` method on embedder can be removed.

Reply:
There is a small issue, if only the embedder is present when creating the Generator, this _weights cannot be used.
https://github.com/SciSharp/LLamaSharp/blob/40ac944fb5b9abc3a3f59ea30d8c724ace8270e3/LLama.KernelMemory/LLamaSharpTextEmbeddingGenerator.cs#L62

Reply:
Ah I see. In that case how about exposing a `LLamaWeights` property on `LLamaEmbedder`? Then you can use that to do the tokenizing. If you do that you'll probably need to delete the two obsolete constructors, they've been marked as obsolete for a while so that's not a problem.

Reply:
> In that case how about exposing a LLamaWeights property on LLamaEmbedder?

It seems like a good way.

Reply:
Due to the absence of the Weights attribute in LLamaEmbedder, I have exposed the Context attribute. The deprecated constructors have also been removed.